x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/guided_onboarding/rules_management_tour.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiTourActions, EuiTourStepProps } from '@elastic/eui';
+import { EuiButton, EuiTourStep } from '@elastic/eui';
+import { noop } from 'lodash';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { of } from 'rxjs';
+import { BulkActionType } from '../../../../../../../common/detection_engine/rule_management/api/rules/bulk_actions/request_schema';
+import { useKibana } from '../../../../../../common/lib/kibana';
+import { useFindRulesQuery } from '../../../../../rule_management/api/hooks/use_find_rules_query';
+import { useExecuteBulkAction } from '../../../../../rule_management/logic/bulk_actions/use_execute_bulk_action';
+import { useRulesTableContext } from '../rules_table_context';
+import * as i18n from './translations';
+import { useIsElementMounted } from './use_is_element_mounted';
+
+export const INSTALL_PREBUILT_RULES_ANCHOR = 'install-prebuilt-rules-anchor';
+export const SEARCH_FIRST_RULE_ANCHOR = 'search-first-rule-anchor';
+
+export interface RulesFeatureTourContextType {
+  steps: EuiTourStepProps[];
+  actions: EuiTourActions;
+}
+
+const GUIDED_ONBOARDING_RULES_FILTER = {
+  filter: '',
+  showCustomRules: false,
+  showElasticRules: true,
+  tags: ['Guided Onboarding'],
+};
+
+export enum GuidedOnboardingRulesStatus {
+  'inactive' = 'inactive',
+  'installRules' = 'installRules',
+  'searchRules' = 'searchRules',
+  'enableRules' = 'enableRules',
+  'completed' = 'completed',
+}
+
+export const RulesManagementTour = () => {
+  const { guidedOnboardingApi } = useKibana().services.guidedOnboarding;
+  const { executeBulkAction } = useExecuteBulkAction();
+  const { actions } = useRulesTableContext();
+
+  const isRulesStepActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', 'rules') ?? of(false),
+    false
+  );
+
+  const { data: onboardingRules } = useFindRulesQuery(
```

Comment:
The idea that only `demoRule` is needed so it's possible to check if that rule is loaded. So the functionality can be moved to a separate hook like `useDemoRule()` which returns `demoRule` and `isLoading` which should be enough to properly determine the `tourStatus`.

Reply:
Yes, I like that idea 👍 We could improve that in a follow-up if you don't mind.

Reply:
Sounds good